### PR TITLE
Indicate Stable Version in Docs Navigation 

### PR DESCRIPTION
**Pull request by akestner**
**1. Issue, if available:**
N/A

**2. Description of changes:**
I've added `(stable)` to the version navigation dropdown to indicate that you're viewing docs for the latest stable version and refactored the javascript that displays the version warning. 

This new JS architecture enables custom javascript to be easily added in the future.

**Screenshots**
karpenter-docs-version

**3. How was this change tested?**
Manually on my local environment

**4. Does this change impact docs?**
- [X] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by bwagner5 — APPROVED** [review]
lgtm

File: website/layouts/partials/navbar-version-selector.html [hunk 1]
```diff
@@ -4,10 +4,12 @@
 	{{ $pageDir = .Site.Params.latest_release_version }}
 {{ end }}
 <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
-	{{ $pageDir }}
+	{{ $pageDir }} {{ if eq (index .Site.Params.versions 0) $pageDir }} (stable) {{ end }}
```
Comment (ellistarn) [inline website/layouts/partials/navbar-version-selector.html:7]:
```suggestion
	{{ $pageDir }} {{ if eq (index .Site.Params.versions 0) $pageDir }} (latest) {{ end }}
```

File: website/layouts/partials/navbar-version-selector.html [hunk 2]
```diff
@@ -4,10 +4,12 @@
...
 </a>
 <div class="dropdown-menu" aria-labelledby="navbarDropdownMenuLink" id="navbarVersionSelector">
-	{{ range .Site.Params.versions }}
-	<a class="dropdown-item" href="/{{ . }}/">{{ . }}</a>
+	{{ range $index, $version := .Site.Params.versions }}
+	<a class="dropdown-item" href="/{{ $version }}" data-docs-version="{{ $version }}">
+		{{ $version }} {{ if eq $index 0 }} (stable) {{ end }}
```
Comment (ellistarn) [inline website/layouts/partials/navbar-version-selector.html:12]:
```suggestion
		{{ $version }} {{ if eq $index 0 }} (latest) {{ end }}
```

**Comment by ellistarn** [comment]
Thoughts on using the term latest, since we don't want to imply that the other releases aren't stable.

**Comment by akestner** [comment]
Yeah, I had been thinking about `latest` as well. Should be all set now

**Review by bwagner5 — APPROVED** [review]
lgtm

**Pull request merged**